app/decorators/physical_server_decorator.rb

```diff
@@ -0,0 +1,9 @@
+class PhysicalServerDecorator < MiqDecorator
+  def fileicon
+    "svg/vendor-#{vendor.try(:downcase)}.svg"
```

Comment:
Yeah, we did it(fix the factory), missing only to push it. But my point is: once a provider don't parse a vendor, the ManageIQ core will crash, which to me looks like a problem. 

We will push it without the factory fixes and remove the `try` as @skateman asked.